Eliminar producto

ramirezyhonny opened the pull request.

mauroandradev committed:
Falta carga imagenes

ger-rinaldi committed:
feat: agregar metodo de serializacion de producto

Permite serializar un objeto Producto a un objeto que sea traducible a
un objeto JavaScript, en este caso un diccionario para que al devolver
la respuesta jsonify lo convierta en un Objeto JS

ger-rinaldi committed:
feat: agregar endpoint leer producto por id

endpoint de url variable que recibe como argumento variable la id del
producto a leer. Devuelve toda la información a excepción del nombre
aleatorio de imagen y la imagen en si.

ger-rinaldi committed:
feat: agregar endpoint busqueda de producto por query

busqueda por consulta GET, recibe nombre, precio_min, precio_max,
exis_min, exis_max para buscar productos que se correspondan con los
valores de dichos campos. De no indicar campo alguno, se devuelven
productos por id.

Permite paginacion.

ger-rinaldi committed:
feat: agregar funcion de carga de imagenes a json

funcion que lee las imagenes del directorio correspondiente, codifica
de binario a utf-8 para agregarlas al json de respuesta.

mauroandradev committed:
Endpoint alta terminado

Tincho25 committed:
actualizar producto

Tincho25 committed:
actualizar producto

Tincho25 committed:
segundo commit: actualizar el nombre de la imagen

Tincho25 committed:
segundo commit: actualizar el nombre de la imagen

Tincho25 committed:
3er commit: actualizacion de variable de entorno

Tincho25 committed:
3er commit: actualizacion de variable de entorno

Tincho25 committed:
4to commit?: probablemente 4to commit

Tincho25 committed:
4to commit?: probablemente 4to commit

ramirezyhonny committed:
cambios_rama_eliminar

ramirezyhonny committed:
cambios_rama_eliminar

ramirezyhonny committed:
commit 1/04/24

ramirezyhonny committed:
commit 1/04/24

ger-rinaldi committed:
fix: logica de actualizacion en endpoint actualizar

Sobrescribir imagen de producto previa, actualizar datos recibidos de usuario.

ger-rinaldi committed:
fix: logica de actualizacion en endpoint actualizar

Sobrescribir imagen de producto previa, actualizar datos recibidos de usuario.

ger-rinaldi commented on product_api_service/api/app_delete.py:
Yhonny, si podes eliminá este archivo en un nuevo commit, ya que este archivo seguro fue para debuggear sería mejor que no pase a main.

ger-rinaldi commented on product_api_service/api/blueprint_elim.py:
```diff
@@ -0,0 +1,34 @@
+
+#endpoint "eliminar producto"
+@blueprint_eliminar.route('/eliminar', methods=["POST"])
+def eliminar_producto():
```
Y acá lo que diría es que la definición del endpoint quede como '/eliminar/<id_producto>' y elimines la linea 12 y 13, así podemos usarlo desde el frontend sin tener que preparar un JSON lo que nos dejaría usarlo haciendo solo un post a 'dominio/api/producto/eliminar/<id_producto>'.

ger-rinaldi reviewed (changes requested):
Yhonny ahí dejé unos comentarios para que mergeemos directo a main despues de que hagas esos cambios, está 10 puntos el endpoint.

ger-rinaldi committed:
Merge pull request #3 from sporter-management/cargarProducto

Merge del primer endpoint carga de producto.

mauroandradev committed:
Merge pull request #2 from sporter-management/leerProductos

Leer productos

ger-rinaldi committed:
fix: registrar endpoints de lectura en blueprint API

mauroandradev committed:
Merge pull request #4 from sporter-management/register_read

fix: registrar endpoints de lectura en blueprint API

ger-rinaldi committed:
feat: agregar Dockerfile para construir imagen de api

ger-rinaldi committed:
fix: agregar dependencias faltantes

- flask-cors para permitir cross origin requests
- mysql-connector como driver de mysql
- gunicorn para servir la aplicacion en produccion

ger-rinaldi committed:
fix: agregar configuracion CORS a flask app

ger-rinaldi committed:
Eliminar product_api_service/api/test

Eliminacion de archivo de desarrollo local

ger-rinaldi committed:
Eliminar product_api_service/api/test

Eliminacion de archivo de desarrollo local

ger-rinaldi committed:
Merge branch 'updateProductoFix' of github.com:sporter-management/product_api_service into updateProductoFix

ger-rinaldi committed:
Merge pull request #5 from sporter-management/updateProductoFix

Update producto fix y conflictos resueltos

ger-rinaldi committed:
Merge branch 'eliminarProducto' of github.com:sporter-management/product_api_service into eliminarProducto

ger-rinaldi reviewed (approved):
Cambios ya efectuados junto con resuloción de conflictos para hacer merge directo.
